rtmp-client/src/main/cpp/librtmp/rtmp.c

```diff
@@ -4975,12 +4975,12 @@ RTMP_Read(RTMP *r, char *buf, int size)
   /* can't continue */
 fail:
   switch (r->m_read.status) {
-  case RTMP_READ_EOF:
-  case RTMP_READ_COMPLETE:
-    return 0;
   case RTMP_READ_ERROR:  /* corrupted stream, resume failed */
     SetSockError(EINVAL);
-    return -1;
+    return r->m_read.status;
```

Comment:
Does `SetSockError` change the value of `r->m_read.status`? If not:

1. Can we just read `r->m_read.status` once into a local variable before the start of the switch, and use it throughout the switch rather than calling is again? It makes it much clearer that the value wont change.
1. What is the purpose of calling `SetSockError`?

Reply:
1. Done. Please take a look.
2. `SetSockError `sets the `errno` variable, which is returned when `GetSockError ` is called.